test/e2e/session_test.go

```diff
@@ -77,13 +75,29 @@ var _ = Describe("Single debug mode", func() {
 	})
 })
 
-func findPod(pods *v1.PodList, deploymentName string) (string, error) {
+func waitForPod(cs *kubernetes.Clientset, testNamespace, deploymentName string) (string, error) {
+	timeLimit := 10
+	timeStepSleepDuration := time.Second
+	for i := 0; i < timeLimit; i++ {
+		pods, err := cs.CoreV1().Pods(testNamespace).List(metav1.ListOptions{})
+		if err != nil {
+			return "", err
+		}
+		if podName, found := findPod(pods, deploymentName); found {
+			return podName, nil
```

Comment:
looks good overall - i would also check that the status is ready. otherwise test might be flaky